File: javascript/7_kyu/exclamation_mark_series_3.js
// Exclamation marks series #3: Remove all exclamation marks from sentence except at the end
// https://www.codewars.com/kata/57faefc42b531482d5000123/train/javascript

function remove(s) {
  var arr = [];

  for (var i = 0; i < s.length; i++) {
    if ( !(s[i] == "!") ) {
      arr.push(s[i])
    }
  }

  var arr_epoint = [];

  for (var i = s.length - 1; i >= 0; i--) {
    if (s[i] == "!") {
      arr_epoint.push(s[i]);
    } else {
      break;
    }
  }

  return arr.join("") + arr_epoint.join("");
}

// // Best Solution

// function remove(s) {
//   return s.replace(/!+(?!!*$)/g, '');
// }

console.log(remove("!Hi!"))

File: python/7_kyu/exclamation_mark_series_3.py
# Exclamation marks series #3: Remove all exclamation marks from sentence except at the end
# https://www.codewars.com/kata/57faefc42b531482d5000123/train/python

def remove(s):
  list_s = list(s)
  arr = []
  arr_epoint = []

  for i in list_s:
    if i != "!":
      arr.append(i)
  
  for i in list_s[::-1]:
    if i == "!":
      arr_epoint.append(i)
    else:
      break

  return ''.join(arr) + ''.join(arr_epoint)

# # Best Solution
# def remove(s):
#     return s.replace('!', '')+ '!'*(len(s)- len(s.rstrip('!')))

print(remove("!Hi!"))